### PR TITLE
T/mforets/11

### PR DESCRIPTION
**Pull request by mforets**

### CODE REVIEW

File: src/solve.jl [hunk 1]
```diff
@@ -147,32 +182,30 @@ end
...
-        options[:plot_vars],
-        n,
-        options[:δ],
-        Rsets,
-        options[:algorithm],
-        ɛ=options[:ɛ],
-        transformation_matrix=transformation_matrix,
-        projection_matrix=options[:projection_matrix]
-        )
+                 options::Options; transformation_matrix=nothing)
```
Comment (schillic) [inline src/solve.jl]: Why is `transformation_matrix` excluded from the doc and still kept as an argument?

**Comment by schillic** [comment]
I do not see the benefit of having two types for 2D and ND solutions, but ok.

I would have done it like that:
```julia
struct ReachSolution{T::LazySet}
   `Xk::Vector{T}`
   ...
end
```

**Comment by mforets** [comment]
> I do not see the benefit of having two types for 2D and ND solutions, but ok.

at some point i thought this would be "needed", but actually not really. in fact with your solution, we can dispatch the plot recipe like `sol::ReachSolution{HPolygon}`.

File: src/plot_recipes.jl [hunk 2]
```diff
@@ -1,4 +1,19 @@
 """
+    plot_sol(sol::RsetsNDSolution; ...)
+
+Plots the solution of a reachability problem in ND.
+
+### Input
+
+- `RsetsNDSolution` -- the solution of a reachability problem in high-dimensions
```
[the remark this answers is not in the document]
Comment (mforets) [reply src/plot_recipes.jl]: thanks for these corrections. i think i'll take your approach and modify these `plot_sol` functions accordingly

File: src/solve.jl (hunk 1 above)
Comment (mforets) [reply src/solve.jl]: i forgot, thanks for pointing this out

**Comment by mforets** [comment]
done

File: src/plot_recipes.jl (hunk 2 above)
[the remark this answers is not in the document]
Comment (schillic) [reply src/plot_recipes.jl]: This and the remark above are still present.

Search for "RsetsSolution".

**Comment by mforets** [comment]
side comment: the idea behind the renaming `AbstractSystem` is to start considering new subtypes that could be `ContinuousAffineODESystem`, `ContinuousPolynomialODESystem`, `ContinuousHybridSystem`, and such.

File: src/plot_recipes.jl (hunk 2 above)
[the remark this answers is not in the document]
Comment (mforets) [reply src/plot_recipes.jl]: thanks for pointing it out, i fixed it now

File: src/solve.jl (hunk 1 above)
Comment (schillic) [reply src/solve.jl]: I still see two "RsetsSolution"s ;)

Comment (mforets) [reply src/solve.jl]: line num?

Comment (mforets) [reply src/solve.jl]: fixed

Comment (schillic) [reply src/solve.jl]: `plot_recipes.jl`, l. 2 and 17

**Pull request merged**